### PR TITLE
fix for sourcing rosfish

### LINKED ISSUE #261: rosfish from noetic release contains bash code and can't be sourced
https://github.com/ros/ros/pull/254 introduced bash code into the rosfish and it results in the following error:
```
/opt/ros/noetic/share/rosbash/rosfish (line 265): Unsupported use of '='. In fish, please use 'set pic "${rosvals[2]}${rosvals[3]}${rosvals[4]}"'.
from sourcing file /opt/ros/noetic/share/rosbash/rosfish
source: Error while reading file “/opt/ros/noetic/share/rosbash/rosfish”
```
Fixing this error, unveils more bash style code. 

P.S.
I'm trying to fix it now, but I don't have much experience with fish scripting

### PR DESCRIPTION
**Pull request by iPaddy**
Fixes #261 
Fixed the script such that it sources in fish.
Functionality should match the intentions of the bash part.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Thanks for the patch.

I updated the ticket description to use the GitHub keyword "Fixes #261" to automatically close the referenced issue when this PR is merged.

File: tools/rosbash/rosfish [hunk 1]
```diff
@@ -22,7 +22,7 @@ function _rosfind
         # BSD find needs -E for extended regexp
         find -E $argv
     else
-        find $argv
+        find $arg
```
Comment (kenji-miyake) [inline tools/rosbash/rosfish:25]: @iPaddy Excuse me, why was this changed? I believe `$arg` is empty here and `$argv` is correct.

How to confirm:

```sh
# Create directory to avoid slow `find`
mkdir tmp
cd tmp

# Shows no candidate
rosrun rviz <Tab>

# Edit `find $arg` to `find $argv`
sudo vim /opt/ros/noetic/share/rosbash/rosfish

# Correctly be complemented
rosrun rviz <Tab>
```

Comment (eduidl) [reply tools/rosbash/rosfish:25]: I agree with you. I created PR.

https://github.com/ros/ros/pull/279
